Assets/Scripts/Core.cs

```diff
@@ -64,31 +85,42 @@ public class Core : MonoBehaviour {
             }
         }
 
-        foreach(string s in cfgTxt.Split('\n')) {
+        foreach (string s in cfgTxt.Split('\n')) {
             string[] properties = s.Split('=');
-            if(properties.Length == 2) {
+            if (properties.Length == 2) {
                 configs.Add(properties[0], properties[1]);
             }
         }
     }
 
     void FixedUpdate() {
-        if(mapRenderer.Ready) {
+        if (mapRenderer.Ready) {
             mapRenderer.FixedUpdate();
         }
     }
 
     void Update() {
-        mapDropdown.gameObject.SetActive(Cursor.lockState != CursorLockMode.Locked);
-        if(mapRenderer.Ready) {
+        //mapDropdown.gameObject.SetActive(Cursor.lockState != CursorLockMode.Locked);
+        if (mapRenderer.Ready) {
             mapRenderer.Render();
         }
+
+        RaycastHit hit;
```

Comment:
That brings one question which may divide opinions, which is better?

- A single `Raycast` and multiple `Action` to broadcast that
- Each component deals with its own `Raycast` stuff

Reply:
First option seems more reasonable.
I see your point. What this raycast actually is, is a mouse to world position converter.
So maybe it would make sense to make a Mouse class and do this raycasting there (a single one), with a way to register the two types of actions we need: OnMouseRaycast and OnMouseClicked .
Then pathfinding and grid selector would register themselves to these actions. Does this make sense?

Reply:
We could have multiple Raycasts, but since both use cases we have right now use the exactly same raycast, The only reason not to optimize it would be to have a cleaner code base. Then grid selector and pathfinding would each have their own things, neatly separated from each other.
I'm not strongly in favor of any of them, you can choose what is more feasible. (copied from discord)